runtime/testy/runtime.py

```diff
@@ -42,31 +45,37 @@ def runtime():
     spawnProcess(PROCESS_NAMES.UDP_RECEIVE_PROCESS, startUDPReceiver)
     spawnProcess(PROCESS_NAMES.HIBIKE, startHibike)
     while True:
-      if restartCount >= 5 or emergency_stopped:
-        print(RUNTIME_CONFIG.DEBUG_DELIMITER_STRING.value)
-        print("Too many restarts, terminating")
+      if restartCount >= 3 or emergency_stopped:
+        if not testMode:
+          print(RUNTIME_CONFIG.DEBUG_DELIMITER_STRING.value)
+          print("Too many restarts, terminating")
         break
-      print(RUNTIME_CONFIG.DEBUG_DELIMITER_STRING.value)
-      print("Starting studentCode attempt: %s" % (restartCount,))
-      spawnProcess(PROCESS_NAMES.STUDENT_CODE, runStudentCode)
+      if not testMode:
+        print(RUNTIME_CONFIG.DEBUG_DELIMITER_STRING.value)
+        print("Starting studentCode attempt: %s" % (restartCount,))
+      spawnProcess(PROCESS_NAMES.STUDENT_CODE, runStudentCode, testName, maxIter)
       while True:
         newBadThing = badThingsQueue.get(block=True)
         print(newBadThing.event)
-        print(newBadThing.data)
+        if not testMode:
+          print(newBadThing.data)
         if newBadThing.event in restartEvents:
-          if(not emergency_stopped and newBadThing.event is BAD_EVENTS.EMERGENCY_STOP):
+          if (not emergency_stopped and newBadThing.event is BAD_EVENTS.EMERGENCY_STOP):
             emergency_stopped = True #somehow kill student code using other method? right now just restarting on e-stop
           break
       stateQueue.put([SM_COMMANDS.RESET, []])
-      terminate_process(allProcesses[PROCESS_NAMES.STUDENT_CODE])
```

Comment:
I moved the dictionary lookup into the terminate_process function. This way i can simultaenously remove a terminated process from all processes. 
Before, this code would just overwrite the old value in the dictionary when the new process is started.